File: app/src/main/java/com/example/android/learnarchitecturecomponents/room/dao/FruitDao.java
package com.example.android.learnarchitecturecomponents.room.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.example.android.learnarchitecturecomponents.room.entities.Fruit;

import java.util.List;

@Dao
public interface FruitDao {

    @Insert(onConflict = OnConflictStrategy.ABORT)
    long insertFruit(Fruit fruit);

    @Insert(onConflict = OnConflictStrategy.ABORT)
    void insertFruits(List<Fruit> fruitList);

    @Query("SELECT * FROM fruit")
    LiveData<List<Fruit>> getFruits();
}

File: app/src/main/java/com/example/android/jetpackdemo/room/dao/FruitDao.kt
package com.example.android.jetpackdemo.room.dao

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

import com.example.android.jetpackdemo.room.entities.Fruit

@Dao
interface FruitDao {

    @get:Query("SELECT * FROM fruit")
    val fruits: LiveData<List<Fruit>>

    @Query("SELECT * FROM fruit WHERE id= :id LIMIT 1")
    fun queryFruitWithId(id: Long): LiveData<Fruit?>

    @Insert(onConflict = OnConflictStrategy.ABORT)
    suspend fun insertFruit(fruit: Fruit)

    @Insert(onConflict = OnConflictStrategy.ABORT)
    suspend fun insertFruits(fruitList: List<Fruit>)
}
